### PR TITLE
Support tier1-segment port as the node port

### PR DESCRIPTION
**Pull request by heypnus**
Tier1-segment means the segment who has path
`/infra/tier-1s/<tier-1-id>/segments/<segment-id>`.

Previously we had already supported the infra-segment,
whose path is` /infra/segments/<segment-id>`.

### CODE REVIEW

File: pkg/controller/node/node_controller.go [hunk 1]
```diff
@@ -461,8 +462,15 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 	// TODO: find a way to fill the segmentPort by using the return value from searchNodePortByVcNameAddress then we won't need to invoke the Get API again.
 	sl := strings.Split(*segmentPort.Path, "/")
 	segmentId := sl[len(sl)-1]
-	portsClient := segments.NewDefaultPortsClient(nsxClients.PolicyConnector)
-	*segmentPort, err = portsClient.Get(segmentId, *segmentPort.Id)
+	portsClient := infra_segment.NewDefaultPortsClient(nsxClients.PolicyConnector)
```
Comment (dantingl) [inline pkg/controller/node/node_controller.go:465]: Move line 465 to 472?
Please add the path pattern for infra segment in comments as well

Comment (heypnus) [reply pkg/controller/node/node_controller.go:465]: > Move line 465 to 472?
Previous this line locates at 472, but it reported an error that portsClient is undefined, so I moved it to 465
> Please add the path pattern for infra segment in comments as well
Done

Comment (dantingl) [reply pkg/controller/node/node_controller.go:465]: Move 465 to 472, and in 470 define portsClient as well
portsClient := tier1_segment.NewDefaultPortsClient(nsxClients.PolicyConnector)

File: pkg/controller/node/node_controller.go [hunk 2]
```diff
@@ -503,7 +515,11 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 		var clusterTag = model.Tag{Scope: &clusterScope, Tag: &cluster}
 		segmentPort.Tags = append(segmentPort.Tags, clusterTag)
 	}
-	_, err = portsClient.Update(segmentId, *segmentPort.Id, *segmentPort)
```
Comment (dantingl) [inline pkg/controller/node/node_controller.go:506]: Sorry I did not know we still need portsClient here

**Review by dantingl — APPROVED** [review]

File: pkg/controller/node/node_controller.go (hunk 2 above)
Comment (heypnus) [reply pkg/controller/node/node_controller.go:506]: Yes and I found the Update methods from infra_segment and tier1_segment are different so in the latest change I split the clients to infraPortsClient and tier1PortsClient

**Review by ggverma — APPROVED** [review]

**Pull request merged**